updated CSF defaults for cyberpanel

meramsey opened the pull request:
Disable unnecessary alerts and unused functions. Enable other functions. Map log universal logs files to the proper location for cyberpanel. This will save a lot of frustration upon initial install with good defaults that aren't too aggressive and cause conflicts.

@usmannasir  If there is a way to have an additional OS-specific version set of commands run on CSF install based on OS version the below also needs to be used so the logs LFD and CSF are mapped properly.  I ran out of steam and not sure how best to detect that so that might be best done by someone else logic wise.

For Ubuntu base Cyberpanel with CSF
sed -i 's|^SSHD_LOG =.*|SSHD_LOG = "/var/log/auth.log"|g' /etc/csf/csf.conf
sed -i 's|^SU_LOG =.*|SU_LOG = "/var/log/auth.log"|g' /etc/csf/csf.conf
sed -i 's|^FTPD_LOG =.*|FTPD_LOG = "/var/log/auth.log"|g' /etc/csf/csf.conf
sed -i 's|^SMTPAUTH_LOG =.*|SMTPAUTH_LOG = "/var/log/mail.log"|g' /etc/csf/csf.conf
sed -i 's|^POP3D_LOG =.*|POP3D_LOG = "/var/log/mail.log"|g' /etc/csf/csf.conf
sed -i 's|^IMAPD_LOG =.*|IMAPD_LOG = "/var/log/mail.log"|g' /etc/csf/csf.conf
sed -i 's|^IPTABLES_LOG =.*|IPTABLES_LOG = "/var/log/kern.log"|g' /etc/csf/csf.conf
sed -i 's|^SYSLOG_LOG =.*|SYSLOG_LOG = "/var/log/syslog"|g' /etc/csf/csf.conf

For Centos based Cyberpanel with CSF

sed -i 's|^SSHD_LOG =.*|SSHD_LOG = "/var/log/secure"|g' /etc/csf/csf.conf
sed -i 's|^SU_LOG =.*|SU_LOG = "/var/log/secure"|g' /etc/csf/csf.conf
sed -i 's|^FTPD_LOG =.*|FTPD_LOG = "/var/log/messages"|g' /etc/csf/csf.conf
sed -i 's|^POP3D_LOG =.*|POP3D_LOG = "/var/log/maillog"|g' /etc/csf/csf.conf
sed -i 's|^SMTPAUTH_LOG =.*|SMTPAUTH_LOG = "/var/log/maillog"|g' /etc/csf/csf.conf
sed -i 's|^IMAPD_LOG =.*|IMAPD_LOG = "/var/log/maillog"|g' /etc/csf/csf.conf
sed -i 's|^IPTABLES_LOG =.*|IPTABLES_LOG = "/var/log/messages"|g' /etc/csf/csf.conf
sed -i 's|^SYSLOG_LOG =.*|SYSLOG_LOG = "/var/log/messages"|g' /etc/csf/csf.conf

meramsey committed:
updated CSF defaults for cyberpanel

Disable unnecessary alerts and unused functions. Enable other functions. Map log universal logs files to the proper location for cyberpanel. This will save a lot of frustration upon initial install with good defaults that aren't too aggressive and cause conflicts.

@usmannasir  If there is a way to have an additional OS-specific version set of commands run on CSF install based on OS version the below also needs to be used so the logs LFD and CSF are mapped properly.  I ran out of steam and not sure how best to detect that so that might be best done by someone else logic wise.

For Ubuntu base Cyberpanel with CSF
sed -i 's|^SSHD_LOG =.*|SSHD_LOG = "/var/log/auth.log"|g' /etc/csf/csf.conf
sed -i 's|^SU_LOG =.*|SU_LOG = "/var/log/auth.log"|g' /etc/csf/csf.conf
sed -i 's|^FTPD_LOG =.*|FTPD_LOG = "/var/log/auth.log"|g' /etc/csf/csf.conf
sed -i 's|^SMTPAUTH_LOG =.*|SMTPAUTH_LOG = "/var/log/mail.log"|g' /etc/csf/csf.conf
sed -i 's|^POP3D_LOG =.*|POP3D_LOG = "/var/log/mail.log"|g' /etc/csf/csf.conf
sed -i 's|^IMAPD_LOG =.*|IMAPD_LOG = "/var/log/mail.log"|g' /etc/csf/csf.conf
sed -i 's|^IPTABLES_LOG =.*|IPTABLES_LOG = "/var/log/kern.log"|g' /etc/csf/csf.conf
sed -i 's|^SYSLOG_LOG =.*|SYSLOG_LOG = "/var/log/syslog"|g' /etc/csf/csf.conf

For Centos based Cyberpanel with CSF

sed -i 's|^SSHD_LOG =.*|SSHD_LOG = "/var/log/secure"|g' /etc/csf/csf.conf
sed -i 's|^SU_LOG =.*|SU_LOG = "/var/log/secure"|g' /etc/csf/csf.conf
sed -i 's|^FTPD_LOG =.*|FTPD_LOG = "/var/log/messages"|g' /etc/csf/csf.conf
sed -i 's|^POP3D_LOG =.*|POP3D_LOG = "/var/log/maillog"|g' /etc/csf/csf.conf
sed -i 's|^SMTPAUTH_LOG =.*|SMTPAUTH_LOG = "/var/log/maillog"|g' /etc/csf/csf.conf
sed -i 's|^IMAPD_LOG =.*|IMAPD_LOG = "/var/log/maillog"|g' /etc/csf/csf.conf
sed -i 's|^IPTABLES_LOG =.*|IPTABLES_LOG = "/var/log/messages"|g' /etc/csf/csf.conf
sed -i 's|^SYSLOG_LOG =.*|SYSLOG_LOG = "/var/log/messages"|g' /etc/csf/csf.conf

spyngamerman commented:
See this thread for more explanation:
https://forums.cyberpanel.net/discussion/1986/csf-settings-and-log-paths-for-lfd-integration-from-the-cli#latest

meramsey committed:
Update csf.py

meramsey committed:
Update csf.py

Fix the formatting so it does not create duplicate for some lines due to how it was parsed.
